packages/connect-query/src/utils.test.ts

```diff
@@ -79,57 +75,112 @@ describe("isAbortController", () => {
-
-    const result = safeUpdater(wrappedInput);
-    expectTypeOf(result).toEqualTypeOf<CountResponse>();
-    expect(result).not.toStrictEqual(wrappedInput);
-
-    expectTypeOf(result.count).toEqualTypeOf<bigint>();
-    expect(wrappedInput.count).toStrictEqual(1n);
-    expect(result.count).toStrictEqual(2n);
-    expect(result).toStrictEqual(wrappedOutput);
-    expect(result).toHaveProperty("$typeName");
+describe("createProtobufSafeUpdater", () => {
+  describe("with update message", () => {
+    const schema = { output: Proto2MessageSchema };
+    const update = create(Proto2MessageSchema, {
+      int32Field: 999,
+    });
+    const safeUpdater = createProtobufSafeUpdater(schema, update);
+    it("returns update message for previous value undefined", () => {
+      const next = safeUpdater(undefined);
+      expect(next).toBe(update);
+    });
+    it("returns update message for previous value", () => {
+      const prev = create(Proto2MessageSchema, {
+        int32Field: 123,
+      });
+      const next = safeUpdater(prev);
+      expect(next).toBe(update);
+    });
   });
 
-  it("handles a function updater", () => {
-    const updater = vi.fn(() => create(methodInfo.output, { count: 2n }));
-    const safeUpdater = createProtobufSafeUpdater(methodInfo, updater);
-
-    expectTypeOf(safeUpdater).toEqualTypeOf<
-      (prev?: CountResponse) => CountResponse
-    >();
-    expect(typeof safeUpdater).toStrictEqual("function");
-
-    const result = safeUpdater(wrappedInput);
-    expect(updater).toHaveBeenCalledWith(wrappedInput);
-    expectTypeOf(result).toEqualTypeOf<CountResponse>();
-    expect(result).not.toStrictEqual(wrappedInput);
-
-    expectTypeOf(result.count).toEqualTypeOf<bigint>();
+  describe("with update message init", () => {
+    const schema = { output: Proto2MessageSchema };
+    const update = {
+      int32Field: 999,
+    };
+    const safeUpdater = createProtobufSafeUpdater(schema, update);
+    it("returns update message for previous value undefined", () => {
+      const next = safeUpdater(undefined);
+      expect(next?.int32Field).toBe(999);
+    });
+    it("returns update message for previous value", () => {
+      const prev = create(Proto2MessageSchema, {
+        int32Field: 123,
+      });
+      const next = safeUpdater(prev);
+      expect(next?.$typeName).toBe(Proto2MessageSchema.typeName);
+      expect(next?.int32Field).toBe(999);
+    });
+  });
 
-    expect(wrappedInput.count).toStrictEqual(1n);
-    expect(result.count).toStrictEqual(2n);
-    expect(result).toStrictEqual(wrappedOutput);
-    expect(result).toHaveProperty("$typeName");
+  describe("with updater function", () => {
+    const schema = { output: Proto2MessageSchema };
+    const safeUpdater = createProtobufSafeUpdater(schema, (prev) => {
+      if (prev === undefined) {
+        return undefined;
+      }
+      return {
+        ...prev,
+        int32Field: 999,
+      };
+    });
```

Comment:
We have to explicitly handle the `undefined` case now. In this example, we do not update query data if it does not exist. 

If the goal is to create query data if none exists, the function could be updated to:

```ts
const safeUpdater = createProtobufSafeUpdater(schema, (prev) => {
  prev ??= create(schema.output);
  return {
    ...prev,
    int32Field: 999,
  };
});
```

We could move this logic into `createProtobufSafeUpdater`, and always provide data to the updater function. But then it isn't possible to update only existing query data. I'm not sure what's idiomatic here. WDYT, @paul-sachs?

Reply:
I think this new version is more idiomatic to react-query so I'm good with this updated behaviour.